crates/swc_macro_wasm/src/multi_pass_debug_tests.rs

```diff
@@ -0,0 +1,306 @@
+    println!("WEBPACK ONLY (NO MACROS) TEST:");
+    println!("Input source:");
+    println!("{}", source);
+    println!("\nAfter optimization:");
+    println!("{}", result);
+    
+    println!("Contains '__webpack_modules__': {}", result.contains("__webpack_modules__"));
+}
+
+/// Test the interaction between macros and webpack modules
+#[test]
+fn test_macro_webpack_interaction() {
+    let source = r#"
+        /*@:if condition="INCLUDE_WEBPACK"*/
+        var __webpack_modules__ = {
+            "1": function() {
+                return "conditional webpack module";
+            }
+        };
+        /*@:endif*/
+        
+        let other = "other code";
+        "#;
+
+    let config = json!({
+        "INCLUDE_WEBPACK": false
+    });
+
+    let result = optimize(source.to_string(), config);
+    println!("MACRO-WEBPACK INTERACTION TEST:");
+    println!("Input source:");
+    println!("{}", source);
+    println!("\nAfter optimization:");
+    println!("{}", result);
+    
+    println!("Contains '__webpack_modules__': {}", result.contains("__webpack_modules__"));
+    println!("Contains 'other': {}", result.contains("other"));
+}
+
+/// Test variable usage after macro removal
+#[test]
+fn test_variable_usage_after_macro_removal() {
+    let source = r#"
+        let sharedVar = "shared";
+        
+        /*@:if condition="USE_SHARED"*/
+        console.log(sharedVar);
+        /*@:endif*/
+        
+        let independentVar = "independent";
+        console.log(independentVar);
+        "#;
+
+    let config = json!({
+        "USE_SHARED": false
+    });
+
+    let result = optimize(source.to_string(), config);
+    println!("VARIABLE USAGE AFTER MACRO REMOVAL TEST:");
+    println!("Input source:");
+    println!("{}", source);
+    println!("\nAfter optimization:");
+    println!("{}", result);
+    
+    println!("Contains 'sharedVar': {}", result.contains("sharedVar"));
+    println!("Contains 'independentVar': {}", result.contains("independentVar"));
+    
+    // sharedVar should be removed because its only usage is in a removed macro block
+}
+
+/// Test what happens with nested function calls after macro removal
+#[test]
+fn test_nested_function_calls_after_macro() {
+    let source = r#"
+        function level1() {
+            return level2();
+        }
+        
+        function level2() {
+            return level3();
+        }
+        
+        function level3() {
+            return "deep result";
+        }
+        
+        /*@:if condition="USE_DEEP_CALL"*/
+        let result = level1();
+        /*@:endif*/
+        "#;
+
+    let config = json!({
+        "USE_DEEP_CALL": false
+    });
+
+    let result = optimize(source.to_string(), config);
+    println!("NESTED FUNCTION CALLS AFTER MACRO TEST:");
+    println!("Input source:");
+    println!("{}", source);
+    println!("\nAfter optimization:");
+    println!("{}", result);
+    
+    println!("Contains 'level1': {}", result.contains("level1"));
```

Comment:
It would be better to convert this to an assertion like

```suggestion
    assert!(!result.contains("level1"), "Should not contains 'level1'");
```